packages/js/data/src/orders/reducer.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import { Reducer } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import TYPES from './action-types';
+import { Actions } from './actions';
+import { PartialOrder } from './types';
+import { getOrderResourceName, getTotalOrderCountResourceName } from './utils';
+
+export type OrdersState = {
+	orders: Record<
+		string,
+		{
+			data: number[];
+		}
+	>;
+	ordersCount: Record< string, number >;
+	errors: Record< string, unknown >;
+	data: Record< number, PartialOrder >;
+};
+
+const reducer: Reducer< OrdersState, Actions > = (
+	state = {
+		orders: {},
+		ordersCount: {},
+		errors: {},
+		data: {},
+	},
+	payload
+) => {
+	if ( payload && 'type' in payload ) {
+		switch ( payload.type ) {
+			case TYPES.GET_ORDER_SUCCESS:
+				const orderData = state.data || {};
+				return {
+					...state,
+					data: {
+						...orderData,
+						[ payload.id ]: {
+							...( orderData[ payload.id ] || {} ),
+							...payload.order,
+						},
+					},
+				};
+			case TYPES.GET_ORDERS_SUCCESS:
+				const ids: number[] = [];
+				const nextOrders = payload.orders.reduce<
+					Record< number, PartialOrder >
+				>( ( result, order ) => {
+					ids.push( order.id );
+					result[ order.id ] = order;
```

Comment:
When fetching a specific field while not including `id`, the last piece record's information is assigned to all rows presumably because this assigns to IDs.

<img width="739" alt="Screen Shot 2022-05-25 at 10 52 32 AM" src="https://user-images.githubusercontent.com/10561050/170293437-8316dd2d-943d-4cbe-9ba6-ce678ab62718.png">
<img width="573" alt="Screen Shot 2022-05-25 at 10 50 52 AM" src="https://user-images.githubusercontent.com/10561050/170293609-3db92048-0ca1-45e1-91c4-d0ef88dc71b2.png">

Reply:
The fix for this has been addressed in https://github.com/woocommerce/woocommerce/pull/33063/commits/153d49771742d272a733ac62d0eeb2c2846d869a